docs/network_architecture.rst

```diff
@@ -53,6 +53,74 @@ Federated Machine Learning Flow
     b. Resume the task
 7. Once all rounds are completed the *Coordinator* can just exit
 
+Model weights distribution
+--------------------------
+
+Models can be massive (several dozens of megabytes), and protobuf is
+not suited for exchanging such data. Instead, the participants and the
+coordinator use an S3 bucket to exchange their weights. The exact
+mechanism is represented by the sequence diagram below.
+
+At the beginning of a round (1) the selected participants send a
+``StartTrainingRound`` request.
+
+Once it receives a response, the participant fetches the weights for
+the current global model from the S3 store (2). S3 buckets are
+key-value stores, and the key for global weights is
+``<round>/global``.
+
+Then, the participant trains. Once done, it uploads its local weights
+to the S3 bucket (3). The key is ``<round_number>/<participant_id>``.
```

Comment:
both primary keys should be called the same, either `<round>` or `<round_number>`